src/redux/game/saga.ts

```diff
@@ -32,7 +32,9 @@ export default function* gameSaga(gameEngine: GameEngine) {
 
     switch (action.type) {
       case messageActions.MESSAGE_RECEIVED:
-        newState = gameEngine.receivePosition(positionFromHex(action.message));
+        yield put(walletActions.decodeStateRequest(action.message));
+        const decodeAction: walletActions.DecodeStateSuccess = yield take(walletActions.DECODE_STATE_SUCCESS);
+        newState = gameEngine.receivePosition(positionFromHex(decodeAction.state,action.message));
```

Comment:
I see you've split the decoding between the wallet and the app, so that the wallet decodes the part of the state that it knows about and the app decodes the application-specific state. The way you've done this introduces a run-time dependency between these two components, so that now the app is unable to decode a state without calling out to the wallet. I wondered if it would be possible to remove this run-time dependency by instead making a code dependency between these two components e.g. the app imports `decode` from the wallet, allowing it to build its own app-state-aware `decode` method on top. This would potentially fit with a long-term pattern where the shared state decoding logic is contained in `fmg-core` and imported by both the wallet and the app. (It might even be worth refactoring to this now.)

Reply:
That's a good point. I'll put change the `game` state decode to just call through to the wallet state and put in an issue to pull it out to `fmg-core` proper.